x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -40,34 +44,30 @@
  * A permission that is based on privileges for index related actions executed
  * on specific indices
  */
-public final class IndicesPermission implements Iterable<IndicesPermission.Group> {
+public final class IndicesPermission {
```

Comment:
`groups()` is sufficient to iterate over the groups (and that is preferable because it expresses ordering that is required in tests)